src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -21,7 +22,9 @@ public class ContentResult : ActionResult
 
             if (!string.IsNullOrEmpty(ContentType))
             {
-                response.ContentType = ContentType;
+                var contentTypeHeader = new MediaTypeHeaderValue(ContentType);
+                contentTypeHeader.Encoding = ContentEncoding;
```

Comment:
Yes, default to UTF-8. You should also be calling the overload of WriteAsync that takes an encoding or else UTF-8 will always be used.